pandas/io/pytables.py

```diff
@@ -2861,7 +2861,7 @@ def write(self, obj, **kwargs):
         super(BlockManagerFixed, self).write(obj, **kwargs)
         data = obj._data
         if not data.is_consolidated():
-            data = data.consolidate()
+            data = data._consolidate()
```

Comment:
Here, `data` is not a DataFrame, but a BlockManager, so this should be left as `consolidate`